src/PostgREST/CLI.hs

```diff
@@ -0,0 +1,351 @@
+    hSetBuffering stderr NoBuffering
+
+    -- read PGRST_ env variables
+    env <- readEnvironment
+
+    -- read path from commad line
+    opts <- readCLIShowHelp env
+
+    -- build the 'AppConfig' from the config file path
+    conf <- readValidateConfig env $ configPath opts
+
+    -- These are config values that can't be reloaded at runtime. Reloading some of them would imply restarting the web server.
+    let
+      host = configServerHost conf
+      port = configServerPort conf
+      maybeSocketAddr = configServerUnixSocket conf
+      dbUri = toS (configDbUri conf)
+      (dbChannelEnabled, dbChannel) = (configDbChannelEnabled conf, toS $ configDbChannel conf)
+      serverSettings =
+          setHost ((fromString . toS) host) -- Warp settings
+        . setPort port
+        . setServerName (toS $ "postgrest/" <> prettyVersion) $
+        defaultSettings
+      poolSize = configDbPoolSize conf
+      poolTimeout = configDbPoolTimeout' conf
+      logLevel = configLogLevel conf
+
+    case command opts of
+      CmdDumpConfig ->
+        do
+          putStr $ dumpAppConfig conf
+          exitSuccess
+      CmdDumpSchema ->
+        do
+          dumpedSchema <- dumpSchema conf
+          putStrLn dumpedSchema
+          exitSuccess
+      CmdRun ->
+        pass
+
+    -- create connection pool with the provided settings, returns either a 'Connection' or a 'ConnectionError'. Does not throw.
+    pool <- P.acquire (poolSize, poolTimeout, dbUri)
+
+    -- Used to sync the listener(NOTIFY reload) with the connectionWorker. No connection for the listener at first. Only used if dbChannelEnabled=true.
+    mvarConnectionStatus <- newEmptyMVar
+
+    -- No schema cache at the start. Will be filled in by the connectionWorker
+    refDbStructure <- newIORef Nothing
+
+    -- Helper ref to make sure just one connectionWorker can run at a time
+    refIsWorkerOn <- newIORef False
+
+    -- Config that can change at runtime
+    refConf <- newIORef conf
+
+    -- This is passed to the connectionWorker method so it can kill the main thread if the PostgreSQL's version is not supported.
+    mainTid <- myThreadId
+
+    let connWorker = Connection.worker mainTid pool refConf refDbStructure refIsWorkerOn (dbChannelEnabled, mvarConnectionStatus)
+
+    -- Sets the initial refDbStructure
+    connWorker
+
+    installHandlers opts refConf env mainTid pool connWorker
+
+    -- reload schema cache on NOTIFY
+    when dbChannelEnabled $
+      Connection.listener dbUri dbChannel pool refConf refDbStructure mvarConnectionStatus connWorker
+
+    -- ask for the OS time at most once per second
+    getTime <- mkAutoUpdate defaultUpdateSettings {updateAction = getCurrentTime}
+
+    let postgrestApplication =
+          postgrest
+            logLevel
+            refConf
+            refDbStructure
+            pool
+            getTime
+            connWorker
+
+    runWithSocket conf maybeSocketAddr serverSettings postgrestApplication
```

Comment:
`conf` is only used here to get `configServerUnixSocketMode`. Pass the value of that directly and it will remove another internal dependency (Main->Config, currently).

Reply:
Fixed